tests/conftest.py

```diff
@@ -59,11 +94,30 @@ def pytest_sessionfinish(session: pytest.Session, exitstatus: int) -> None:
     """Remove temporary config directory."""
     shutil.rmtree(nessie_test_config.config_dir)
 
+    if nessie_test_config.nessie_container is not None:
+        nessie_test_config.nessie_container.stop()
+
+
+def pytest_runtest_setup(item: pytest.Item) -> None:
+    """Starts a Nessie container, if needed."""
+    if nessie_test_config.nessie_container is None and item.get_closest_marker("nessie") is not None:
+        nessie_test_config.nessie_container = NessieContainer(image=NessieContainer.DEFAULT_TEST_IMAGE)
```

Comment:
is `DEFAULT_TEST_IMAGE` the latest "snapshot release" of the nessie image?

because `DEFAULT_IMAGE` is the latest official release (which we dont use in these tests now it seems).

i am wondering whether this may cause issues in the following cases:
- snapshot is broken for some reason (suddenly pynessie PRs cant pass CI)
- snapshot has an intentional breaking change (suddenly pynessie PRs cant pass CI even though they might be unrelated)

so just thinking out loud: maybe PR CI should run against a known good image version and on `main` branch we have scheduled CI run against unstable

but we can also figure out the issues once they actually happen

Reply:
thinking about this again:

it may also be a question about compatibility guarantees i.e. what pynessie versions are working with what nessie server versions?
if we have some backwards compatibility guarantees we might want to run against those versions in CI.

previously both were released at the same time (but potentially broken due to VCR stuff not getting updated) but now with a separate repos (and versions?) it may be even less clear

Reply:
The reason was actually the 2nd point: identify breaking changes w/ Nessie snapshot releases.

I was quite surprised how much stuff was actually no longer working - were just/mostly tests, but still.

Reply:
Yea, we'd need CI against multiple release incl "latest main". But that can be done later, WDYT?

Reply:
yes later is fine